base/server/healthcheck/pki/server/healthcheck/clones/plugin.py

```diff
@@ -0,0 +1,310 @@
+
+from pki.server.healthcheck.core.main import merge_dogtag_config
+
+import logging
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+# Temporary workaround to skip VERBOSE data. Fix already pushed to upstream
+# freeipa-healthcheck: https://github.com/freeipa/freeipa-healthcheck/pull/126
+logging.getLogger().setLevel(logging.WARNING)
+
+
+class ClonesPlugin(Plugin):
+    def __init__(self, registry):
+        # pylint: disable=redefined-outer-name
+        super(ClonesPlugin, self).__init__(registry)
+
+        self.master_cas = []
+        self.clone_cas = []
+        self.master_kras = []
+        self.clone_kras = []
+        self.master_ocsps = []
+        self.clone_ocsps = []
+        self.master_tpss = []
+        self.clone_tpss = []
+        self.master_tkss = []
+        self.clone_tkss = []
+
+        self.security_domain = None
+        self.db_dir = None
+        self.subsystem_token = None
+        self.passwd = None
+
+        self.instance = PKIInstance(self.config.instance_name)
+
+    def contact_subsystem_using_pki(
+            self, subport, subhost, subsystemnick,
+            token_pwd, db_path, cmd, exts=None):
+        command = ["/usr/bin/pki",
+                   "-p", str(subport),
+                   "-h", subhost,
+                   "-n", subsystemnick,
+                   "-P", "https",
+                   "-d", db_path,
+                   "-c", token_pwd,
+                   cmd]
+
+        if exts is not None:
+            command.extend(exts)
+
+        output = None
+        try:
+            output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf-8')
+            return output
+
+        output = output.decode('utf-8')
+
+        return output
+
+    def contact_subsystem_using_sslget(
+            self, port, host, subsystemnick,
+            token_pwd, db_path, params, url):
+
+        command = ["/usr/bin/sslget"]
+
+        if subsystemnick is not None:
+            command.extend(["-n", subsystemnick])
+
+        command.extend(["-p", token_pwd, "-d", db_path])
+
+        if params is not None:
+            command.extend(["-e", params])
+
+        command.extend([
+            "-r", url, host + ":" + port])
+
+        logger.info(' command : %s ', command)
+        output = None
+        try:
+            output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf-8')
+            return output
+
+        output = output.decode('utf-8')
+
+        return output
+
+    def get_security_domain_data(self, host, port):
+        domain_data = None
+
+        try:
+            connection = PKIConnection(protocol='http',
+                                       hostname=host,
+                                       port=port,
+                                       verify=False)
```

Comment:
Since we are trying to get the security domain data, should we enable ssl verification? @cipherboy would not like this being turned off :wink:

Reply:
I will give this a try thanks! I was merely checkout out some other health check code that did this. But since it's http in this case, it may be a mute point.

Reply:
I left as is, mostly because this is http connection.